src/main/webapp/election-list.js

```diff
@@ -0,0 +1,120 @@
+const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July',
+                'August', 'September', 'October', 'November', 'December'];
+
+
+function onElectionListLoad(){
+  let searchParams = new URLSearchParams(window.location.search);
+  let selectedState = searchParams.get("state");
+  if(selectedState != null){
+    document.getElementById('select-state').value = selectedState;
+    listElections();
+  }
+}
+
+/**
+ * Fetch the list of upcoming elections and display relevant ones according
+ * to the state selected by the user.
+ */
+function listElections() {
+  let dropdownSelection = document.getElementById('select-state');
+  let selectedStateId = dropdownSelection.options[dropdownSelection.selectedIndex].value;
+  let selectedStateName = dropdownSelection.options[dropdownSelection.selectedIndex].text;
+  let nationalElections = [];
+  let stateElections = [];
+ 
+  // Registering state selection as a query parameter
+  addQueryParameter("state", selectedStateId);
+
+  if (selectedStateId === undefined || selectedStateName === undefined) {
+    return;
+  }
+
+  fetch('/election')
+    .then(response => response.json())
+    .then((electionList) => {
+      // ocdDivisionId is in the form "ocd-division/country:us/state:<state_code>"
+      // For example, Lousiana's ID is "ocd-division/country:us/state:la"
+      let regex = /ocd\-division\/country\:us(\/(state|district)\:([a-z][a-z]))?/;
+
+      // For every election returned, store it if the state matches or if national election,
+      // to later display the details to the user.
+      electionList.forEach((election) => {
+        let ocdId = election.scope;
+
+        // Match the OCD Division ID with the regular expressions to check if this election
+        // is a state election, district election, or national election.
+        let regexMatch = regex.exec(ocdId);
+
+        if (regexMatch != null && regexMatch[3] == selectedStateId) {
+          stateElections.push(election);
+        } else if (regexMatch != null && regexMatch[1] == undefined) {
+          nationalElections.push(election);
+        }
+      });
+
+      let source = document.getElementById('election-list-template').innerHTML;
+      let template = Handlebars.compile(source);
+      let context = { state: selectedStateName, 
+                      stateElections: stateElections, 
+                      nationalElections: nationalElections };
+
+      let electionListContainerElement = document.getElementById('election-list-content');
+      electionListContainerElement.innerHTML = template(context);
+  });
+  updateLinksWithQueryParams(document.links);
+}
+
+function findElectionNameById(electionId) {
+  fetch('/election')
+    .then(response => response.json())
+    .then((electionList) => {
+      electionList.forEach((election) => {
+        // It's worth noting that, on this branch, the
```

Comment:
Just wanted to note that this is not that efficient (fetching every single election in datastore and then parsing in JS).

I'm ok with it since we should be moving to JSP instead soon.
Some other ways to get around this would be to add support for query parameters on /election GET to filter by electionId etc.  (that way the fetch call only returns 1 election).

Reply:
I see! Let me look into the query parameters for fetch; I'll move to JSP next week for sure!